README
IPC::ShareLite 0.18

DESCRIPTION

  IPC::ShareLite provides a simple interface to shared memory, allowing
  data to be efficiently communicated between processes.  Your operating
  system must support SysV IPC (shared memory and semaphores) in order to
  use this module.   
 
COPYRIGHT & TERMS

  Copyright 1998-2002, Maurice Aubrey <maurice@hevanet.com>. 
  All rights reserved.

  This module is free software; you may redistribute it and/or
  modify it under the same terms as Perl itself.

  THIS SOFTWARE IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER
  EXPRESSED OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DAMAGES RESULTING FROM
  THE USE OF THIS SOFTWARE.                      

PREREQUISITES

  This module requires perl 5.004 or later.

INSTALLATION

  To install this module, move into the directory where this file is
  located and type the following:

        perl Makefile.PL
        make
        make test
        make install

  This will install the module into the Perl library directory.  If 
  you lack sufficient privileges for this, then you can specify an
  alternate directory like this:

        perl Makefile.PL PREFIX=/where/I/want/it/put
        make
        make test
        make install

  See the POD documentation for further details.  Once the module
  is installed, you should be able to read the documentation by
  typing the following from the command-line:

	perldoc IPC::ShareLite


Changes
Revision history for Perl extension IPC::ShareLite.

0.16 2009-03-09
    - Add unicode test.

0.15 2009-03-09
    - Add some logging to help diagnose smoke failures.

0.14 2009-03-07
    - Croak if 4 char key string is not 8-bit clean. See #33672.

0.13 2008-03-09
    - Fixed patching of double quoted strings in Makefile. Affects Win32
      and, probably, VMS.

0.12 2008-02-25
    - Use Devel::CheckLib to verify that we have a C compiler.

0.11 2008-02-25
    - Removed POD coverage test. Dynamic constants make it flaky.
    
0.10 2008-02-24
    - Use Perl's malloc wrappers
    - moved test into t/sharelite.t
    - made test use Test::More
    - removed Configure mechanism
    - assorted minor tidying

0.09 Wed Dec  4 13:51:05 PST 2002
        - fixed bug in size() thanks to Craig Manley <c.manley@skybound.nl>

0.08 Sat Jul 15 03:00:16 PDT 2000
        - added define for PL_sv_undef for older perls
          thanks to Christian Gilmore <Christian_Gilmore@tivoli.com>

0.07 Fri Apr 14 02:56:47 PDT 2000
        - changed sv_undef to PL_sv_undef; should now compile
          under 5.006.  Thanks to Frey Kuo <frey@engineer.com> and
          Leon Brocard <leon@astray.com> 
        - improved performance of _rearrange_args() thanks
          to patch from Sam Tregar <sam@tregar.com>
        - renamed all _func names to sharelite_func due to conflict
          with IRIX libraries. Thanks to John Clutterbuck
          <john.clutterbuck@sbs.siemens.co.uk>

0.06 Tue Jan 18 04:33:57 PST 2000
        - Eliminated some warnings thanks to Jim Mosier <jim.mosier@usa.net>
        - Fix DESTROY bug thanks to Sam Tregar <sam@tregar.com>

0.05 Thu Dec 10 08:00:22 PST 1998
        - adding casting to shmat() calls 
        - was checking shmat() return value for NULL instead of -1
	- removed -O3 from Makefile.PL

0.04 Sun Nov 29 17:38:48 PST 1998
	- now using metaconfig to determine whether semun definition
	  is available -- thanks to Paul J. Schinder <schinder@pobox.com>.

0.03 Thu Nov 19 12:36:06 PST 1998
	- changed shlock() and shunlock() to lock() and unlock().
	  old names were confusing.  Old names remain but are
	  deprecated.
	- added accessor methods for key, exclusive, etc.
	- added _initialize() method
	- added retry for EIDRM error when allocating additional
          segments
	- added version() method for use with per-process caches
	- moved segment state variable from semaphore to shared
          memory (it's faster)
        - modified documentation

0.02  Sat Nov 14 05:20:01 1998 
	- removed signal stuff from tests
	- added destroy() accessor method
	- added check for EINVAL when locking semaphore in new_share()
	- replaced lock_ex() and lock_sh() methods with single shlock()
          method that accepts flock() flags.  Renamed unlock() to
          shunlock().
	- LOCK_EX, LOCK_SH, LOCK_UN, and LOCK_NB now available for import.
	- replaced sv_setpvn() call with sv_usepvn() in read_share()
          XS function to remove an unnecessary mem copy.
	- modified documentation. 

0.01  Mon Nov  9 17:49:02 1998
	- original version; created by h2xs 1.18



config.h
/*
 * This file was produced by running the config_h.SH script, which
 * gets its values from config.sh, which is generally produced by
 * running Configure.
 *
 * Feel free to modify any of this as the need arises.  Note, however,
 * that running config_h.SH again will wipe out any changes you've made.
 * For a more permanent change edit config.sh and rerun config_h.SH.
 *
 * $Id: config_h.SH,v 1.1.1.1 2000/04/14 09:23:54 maurice Exp $
 */

/*
 * Package name      : IPC-ShareLite
 * Source directory  : .
 * Configuration time: Sun 24 Feb 2008 00:03:47 GMT
 * Configured by     : andy
 * Target system     : darwin voodoo.local 9.2.0 darwin kernel version 9.2.0: tue feb 5 16:13:22 pst 2008; root:xnu-1228.3.13~1release_i386 i386 
 */

#ifndef _config_h_
#define _config_h_

/* VOIDFLAGS:
 *	This symbol indicates how much support of the void type is given by this
 *	compiler.  What various bits mean:
 *
 *	    1 = supports declaration of void
 *	    2 = supports arrays of pointers to functions returning void
 *	    4 = supports comparisons between pointers to void functions and
 *		    addresses of void functions
 *	    8 = suports declaration of generic void pointers
 *
 *	The package designer should define VOIDUSED to indicate the requirements
 *	of the package.  This can be done either by #defining VOIDUSED before
 *	including config.h, or by defining defvoidused in Myinit.U.  If the
 *	latter approach is taken, only those flags will be tested.  If the
 *	level of void support necessary is not present, defines void to int.
 */
#ifndef VOIDUSED
#define VOIDUSED 15
#endif
#define VOIDFLAGS 15
#if (VOIDFLAGS & VOIDUSED) != VOIDUSED
#define void int		/* is void to be avoided? */
#define M_VOID			/* Xenix strikes again */
#endif

/* HAS_UNION_SEMUN:
 *	This symbol, if defined, indicates that the union semun is
 *	defined by including <sys/sem.h>.  If not, the user code
 *	probably needs to define it as:
 *	union semun {
 *	    int val;
 *	    struct semid_ds *buf;
 *	    unsigned short *array;
 *	}
 */
#define HAS_UNION_SEMUN	/**/

#endif
